dashboard/src/actions/index.js

```diff
@@ -197,3 +198,196 @@ export const addScratchpadEntry = entry => ({
+        .then(function handleResponse(result) {
+            if (result.uids && result.uids.share) {
+                dispatch(updateShareId(result.uids.share));
+            }
+        })
+        .catch(function(error) {
+            console.log(error.stack);
+            return error.message;
+        })
+        .then(function(errorMsg) {
+            if (errorMsg !== undefined) {
+                dispatch(
+                    saveErrorResponse(
+                        "Got error while saving querying for share: " + errorMsg
+                    )
+                );
+            }
+        });
+};
+
+export const getShareId = (dispatch, getState) => {
+    let query = getState().query.text;
+    if (query === "") {
+        return;
+    }
+    let stringifiedQuery = JSON.stringify(encodeURI(query)),
+        // Considering that index is already set on the pred, schema mutation
+        // should be a no-op. Lets see if we have already stored this query by
+        // performing an exact match.
+        checkQuery = `
+mutation {
+    schema {
+        _share_: string @index(exact) .
+    }
+}
+{
+    query(func:eq(_share_, ${stringifiedQuery})) {
+        _uid_
+    }
+}`;
+
+    timeout(
+        6000,
+        fetch(dgraphQuery(false), {
+            method: "POST",
+            mode: "cors",
+            headers: {
+                Accept: "application/json",
+                "Content-Type": "text/plain"
+            },
+            body: checkQuery
+        })
+            .then(checkStatus)
+            .then(response => response.json())
+            .then(function handleResponse(result) {
+                if (result.query && result.query.length > 0) {
+                    dispatch(updateShareId(result.query[0]["_uid_"]));
+                } else {
+                    // Else do a mutation to store the query.
+                    doShareMutation(dispatch, getState);
+                }
+            })
+    )
+        .catch(function(error) {
+            console.log(error.stack);
+            return error.message;
+        })
+        .then(function(errorMsg) {
+            if (errorMsg !== undefined) {
+                dispatch(
+                    saveErrorResponse(
+                        "Got error while saving querying for share: " + errorMsg
+                    )
+                );
+            }
+        });
+};
+
+export const getQuery = shareId => {
+    return dispatch => {
+        timeout(
+            6000,
+            fetch(dgraphQuery(false), {
+                method: "POST",
+                mode: "cors",
+                headers: {
+                    Accept: "application/json"
+                },
+                body: `{
+                    query(id: ${shareId}) {
+                        _share_
+                    }
+                }`
+            })
+                .then(checkStatus)
+                .then(response => response.json())
+                .then(function handleResponse(result) {
```

Comment:
Any reason the callback is not anonymous (i.e. has name)? Is it for V8 optimization?